entities/user.go

```diff
@@ -123,6 +123,7 @@ func (u *User) getWethAndApproveAtlas(amount *big.Int) {
 	}
 
 	// Approve those WETH for Atlas
+	// Q: approve must be done? if there is weth balance, you still need to approve?
 	approve(WETH_ADDRESS, u.addresses["atlas"], wethNeeded, u.ethClient, u.signer)
```

Comment:
In this example, the user approves Atlas for the exact amount of WETH they are selling. We could make the user approve the maximum number possible the first time, so they would not need to approve anymore in the future, just like some other dApps are doing.

Reply:
sure. 
don't you also need to check if such approval was already granted?

Reply:
Yes indeed.